source/Halibut/Transport/Streams/NetworkTimeoutStream.cs

```diff
@@ -186,22 +138,53 @@ public override void EndWrite(IAsyncResult asyncResult)
 
         public override void Flush() => inner.Flush();
 
-        public override int Read(byte[] buffer, int offset, int count) => inner.Read(buffer, offset, count);
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return CloseOnTimeout(() => inner.Read(buffer, offset, count));
+        }
 
         public override long Seek(long offset, SeekOrigin origin) => inner.Seek(offset, origin);
 
         public override void SetLength(long value) => inner.SetLength(value);
 
-        public override void Write(byte[] buffer, int offset, int count) => inner.Write(buffer, offset, count);
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            CloseOnTimeout(() => inner.Write(buffer, offset, count));
+        }
 
-        public override void WriteByte(byte value) => inner.WriteByte(value);
+        public override void WriteByte(byte value)
+        {
+            CloseOnTimeout(() => inner.WriteByte(value));
+        }
 
 #if !NETFRAMEWORK
         public override void CopyTo(Stream destination, int bufferSize) => inner.CopyTo(destination, bufferSize);
         
-        public override int Read(Span<byte> buffer) => inner.Read(buffer);
+        public override int Read(Span<byte> buffer)
+        {
+            try
+            {
+                return inner.Read(buffer);
+            }
+            catch (Exception ex) when (IsTimeoutException(ex))
+            {
+                inner.Close();
+                throw;
+            }
+        }
```

Comment:
You should use that nice `CloseOnTimeout` overload you just wrote! Same for the `Write` override below

Reply:
You can't as you can't use the byref-like structure (Span and ReadOnlySpan) in a lambda

Reply:
😞